src/interface/cmdTlmServer.c

```diff
@@ -0,0 +1,487 @@
+static int cmdTlmServer_recv(commandStreamStructure_t *pCmdMsg,
+                             struct pollfd *pFds, int timeout) {
+
+    int numEvent = poll(pFds, 1, timeout);
+    if (numEvent <= 0) {
+        // error (-1) or timeout (0)
+        return -1;
+    }
+
+    return recv(pFds->fd, pCmdMsg, sizeof(commandStreamStructure_t), 0);
+}
+
+// Send the last command status in message queue if any.
+// Return 0 if sending successfully. Else, -1.
+static int cmdTlmServer_sendLastCmdStateInMsgQueue(serverInfo_t *pServerInfo) {
+    int bytesReceived = mq_receive(pServerInfo->msgQueueCmdStatus,
+                                   (char *)pServerInfo->pMsgCmdStatus,
+                                   sizeof(commandStatusStructure_t), NULL);
+
+    // Send the message only when there is the available data.
+    // Writing to a closed socket will raise SIGPIPE.
+    // For the refereces of this and how to avoid, follow:
+    // https://newbedev.com/how-to-prevent-sigpipes-or-handle-them-properly
+    // https://stackoverflow.com/questions/26752649/so-nosigpipe-was-not-declared
+    // https://stackoverflow.com/questions/19172804/crash-when-sending-data-without-connection-via-socket-in-linux?rq=1
+    int error = 0;
+    if (bytesReceived > 0) {
+        error = send(pServerInfo->socketConnect, pServerInfo->pMsgCmdStatus,
+                     bytesReceived, MSG_NOSIGNAL);
+        if (error < 0) {
+            // Return if the connection is broken
+            return -1;
+        }
+    }
+
+    return 0;
+}
+
+// Close the connection with the TCP/IP client. This function will reset the
+// file descriptor of 'pollfd' structure. The server will be put into the
+// Disconnected state and wait for the new connection request.
+static void cmdTlmServer_closeConn(serverInfo_t *pServerInfo,
+                                   struct pollfd *pFds) {
+
+    syslog(LOG_NOTICE, "Connection socket being reset in %s server.",
+           pServerInfo->pName);
+    tcpServer_close(pServerInfo->socketConnect);
+
+    pServerInfo->socketConnect = -1;
+    pFds->fd = -1;
+
+    pServerInfo->serverStatus = ServerStatus_Disconnected;
+    syslog(LOG_NOTICE, "The state of %s server is disconnected.",
+           pServerInfo->pName);
+}
+
+// Check the command message (pCmdMsg) is valid or not. The status of command
+// will be filled in pCmdStatus if the command is invalid. This function will
+// check the commander as well.
+// Return true if valid. Else, false.
+static bool cmdTlmServer_isCmdValid(commandStatusStructure_t *pCmdStatus,
+                                    commandStreamStructure_t *pCmdMsg,
+                                    bool isCommander) {
+    char *reasonCmdFail = "";
+    bool isCmdValid = true;
+
+    // Check the commander is from CSC/GUI or not
+    if (isCmdValid && !((pCmdMsg->commander == Commander_GUI) ||
+                        (pCmdMsg->commander == Commander_CSC))) {
+        reasonCmdFail = "Invalid commander";
```

Comment:
What does "Invalid" mean in this case? Please consider "Unrecognized commander; must be one of GUI or CSC". (If this was C++ I'd suggest including the value of `commander` in the message, as well as the values for CSC and GUI. But it's not worth the hassle in C).

Reply:
Updated in the commit: ccc826f. Thanks!